services/tasks-svc/internal/task/aggregate/aggregate.go

```diff
@@ -171,9 +171,11 @@ func (a *TaskAggregate) onSubtaskCreated(evt hwes.Event) error {
 		return err
 	}
 
-	subtask := models.NewSubtask()
-	subtask.ID = subtaskID
-	subtask.Name = payload.Name
+	subtask := &models.Subtask{ID: subtaskID, Name: payload.Name}
+
+	if a.Task.Subtasks == nil {
+		a.Task.Subtasks = make(map[uuid.UUID]models.Subtask)
+	}
```

Comment:
we could keep this, but it's not needed anymore with my proposed change to NewAggregate (I think)

Reply:
Yup